app/controllers/concerns/blacklight/facet.rb

```diff
@@ -20,9 +20,14 @@ def facets_from_request(fields = facet_field_names)
       fields.map { |field| facet_by_field_name(field) }.compact
     end
 
-    def facet_field_names(facet_type = :all)
+    def facet_field_names(requested_facet_type = :default)
+      requested_facet_type ||= :default
       field_configs = blacklight_config.facet_fields.values
-      field_configs = field_configs.select {|fc| fc[:facet_type] == facet_type } unless facet_type == :all
+      field_configs = field_configs.select {|fc|
+        configured_facet_type = fc[:facet_type] || :default
```

Comment:
this could be more clear as a single if/else statement, e.g. 

```
configured_facet_type = if configured_facet_type.respond_to?(:lambda?)
                          configured_facet_type.call(params)
                        else
                          fc[:facet_type] || :default
                        end

```

Reply:
agreed; but let's defer to general strategy re: facet_type